File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/project/get/GetProjectsOperationResult.java
/*
 * Copyright 2013 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.project.get;

import java.util.Map;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.get.GetObjectsOperationResult;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.project.Project;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.project.id.IProjectId;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author pkupczyk
 */
@JsonObject("as.dto.project.get.GetProjectsOperationResult")
public class GetProjectsOperationResult extends GetObjectsOperationResult<IProjectId, Project>
{

    private static final long serialVersionUID = 1L;

    @SuppressWarnings("unused")
    private GetProjectsOperationResult()
    {
    }

    public GetProjectsOperationResult(Map<IProjectId, Project> objectMap)
    {
        super(objectMap);
    }
}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/project/get/GetProjectsOperationResult.js
/**
 * @author pkupczyk
 */
define([ "stjs", "as/dto/common/get/GetObjectsOperationResult" ], function(stjs, GetObjectsOperationResult) {
	var GetProjectsOperationResult = function(objectMap) {
		GetObjectsOperationResult.call(this, objectMap);
	};
	stjs.extend(GetProjectsOperationResult, GetObjectsOperationResult, [ GetObjectsOperationResult ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.project.get.GetProjectsOperationResult';
		prototype.getMessage = function() {
			return "GetProjectsOperationResult";
		};
	}, {});
	return GetProjectsOperationResult;
})